src/plugins/lock/README.md

```diff
@@ -0,0 +1,19 @@
+- infos = Information about the lock plugin is in keys below
+- infos/author = Kurt Micheli <e1026558@student.tuwien.ac.at>
+- infos/licence = BSD
+- infos/needs =
+- infos/provides =
+- infos/placements = pregetstorage postgetstorage presetstorage postcommit postrollback
+- infos/status = maintained global
+- infos/description =
+
+## LOCK ##
+
+This global lock plugin introduces a read lock while `GET` and a read/write lock
+while `SET`.
+
+A semaphore is used for the synchronisation and the implemented algorithm favors the writer,
+because updates should be propagated soon as possible and a `kdbSet` is more expansive than
+a `kdbGet`. This expansiveness comes from merging and conflict solving.
```

Comment:
It makes sense to favor the writers (because readers should get up-to-date information), but I cannot follow your explanation. Do you mean expensive instead of expansive? What has the run-time cost to do with preference?

merging is not a part of `kdbSet`, but validation is.